osf/models/mixins.py

```diff
@@ -1834,3 +2108,69 @@ def expand_registration_responses(self):
 
     class Meta:
         abstract = True
+
+
+class EditableFieldsMixin(TitleMixin, DescriptionMixin, CategoryMixin, ContributorMixin,
+        NodeLicenseMixin, Taggable, TaxonomizableMixin, AffiliatedInstitutionMixin):
```

Comment:
this kinda breaks the mixin pattern -- not that i love mixins to begin with, but given that we're using them, i'd expect a mixin to extend at most one base class.

a mixin-of-mixins makes it harder to trace what should be where (e.g. it's confusing to see all the "override for ContributorMixin" comments on classes that don't [directly] extend `ContributorMixin`)

the cleanest solution would be rethinking our whole data layer to avoid all these single-field mixins, but that aside...

i think we should use `EditableFieldsMixin(models.Model)` and either:
1. accept that some fields might not exist; gracefully skip them in `copy_editable_fields` and/or `set_editable_attribute`
2. require all fields to exist; give a useful error when `copy_editable_fields` is called and some fields are missing
3. require that specific list of sibling mixins; add a metaclass that gives a useful error when a subclass of `EditableFieldsMixin` is created with any required siblings missing

Reply:
Conclusions from the DraftNode Moot Meeting: Assuming it may be reasonably debugged if issues arise, we should stick with this pattern for now. `TitleMixin` and `DescriptionMixin` are both used by `Preprint` which is why they may not be combined into a singular `EditableFieldsMixin`.